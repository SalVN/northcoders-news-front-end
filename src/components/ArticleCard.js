import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

const ArticleCard = function (props) {
  return (
    <div className='box'>
      <article className='media'>
        <div className='media-left'>
          {props.number + 1}
        </div>
        <div className='media-left'>
          <button className='button' onClick={props.voteHandler.bind(this, 'up', props.id)}><i className="fa fa-arrow-up" aria-hidden="true"></i></button>
          <div>{props.votes}</div>
          <button className='button' onClick={props.voteHandler.bind(this, 'down', props.id)}><i className="fa fa-arrow-down" aria-hidden="true"></i></button>
        </div>
        <div className='media-content'>
          <div className='content'>
            <Link to={`/articles/${props.id}`}>
              <h3 className='title is-3'>{props.title}</h3>
            </Link>
            <p>
              <span>added by: {props.author}</span>
              <span>tags: {props.tags}</span>
              <span>comments: {props.comment_count}</span>
            </p>
          </div>
        </div>
      </article>
    </div>
  );
};

ArticleCard.propTypes = {
  title: PropTypes.string.isRequired,
  votes: PropTypes.number.isRequired,
  number: PropTypes.number.isRequired,
  author: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  comment_count: PropTypes.number.isRequired,
  voteHandler: PropTypes.func.isRequired
};


export default ArticleCard;
