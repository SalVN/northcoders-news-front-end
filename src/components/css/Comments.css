.comments-title {
    text-align: center;
    margin-top: 10px;
    font-weight: bold;
}

.comments-title-main {
    color: #c1002e !important;
    margin-right: 10px;
    font-size: 24px;
}

.comments-title-sub {
    font-size: 20px;
    color: grey;
}