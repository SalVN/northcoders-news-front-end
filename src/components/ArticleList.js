import React from 'react';
import ArticleCard from './ArticleCard';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import * as actions from '../actions/actions';

class ArticleList extends React.Component {
  componentDidMount() {
    if (this.props.users && this.props.users.length < 1) {
      this.props.fetchUsers();
    }
  }
  render() {
    const users = this.props.users;
    const voteHandler = this.props.voteArticle;
    return (
      <div id='ArticleList' className='content'>
        {this.props.articles.length > 0 &&
          this.props.articles.map((article, i) => {
            const username = article.created_by;
            let index;
            users &&
              users.forEach((user, i) => {
                if (user.username === username) index = i;
              });
            return (
              <ArticleCard
                key={article._id}
                title={article.title}
                number={i}
                votes={article.votes}
                author={article.created_by}
                id={article._id}
                tags={article.belongs_to}
                comment_count={article.comment_count}
                voteHandler={voteHandler}
                userData={users[index]}
              />
            );
          })
        }
      </div>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    fetchUsers: () => {
      dispatch(actions.fetchUsers());
    }
  };
}

function mapStateToProps(state) {
  return {
    loading: state.users.loading,
    users: state.users.users
  };
}

ArticleList.propTypes = {
  articles: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchUsers: PropTypes.func.isRequired,
  voteArticle: PropTypes.func.isRequired,
  users: PropTypes.array.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticleList);