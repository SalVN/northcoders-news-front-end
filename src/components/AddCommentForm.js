import React from 'react';
import PropTypes from 'prop-types';

import './css/AddCommentForm.css';

class AddCommentForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: ''
        };
        this.handleChange = this.handleChange.bind(this);
    }
    render() {
        return (
            <div>
                {this.props.showForm
                    ?
                    <div className='box box-style box-style-comment box-style-comment'>
                        <article className='media'>
                            <div className='media-left'>
                                <img className='comment-form-image' src={this.props.user.avatar_url} />
                            </div>
                            <div className='media-content'>
                                <div className='content'>
                                    <form onSubmit={this.props.handleSubmit}>
                                        <label className='comment-form-heading' htmlFor="comment-input">Join the conversation:</label>
                                        <br />
                                        <div className='comment-form' onBlur={this.props.toggleForm.bind(this, this.state.text)}>
                                            <textarea className='add-comment-textarea' onChange={this.handleChange} id="comment-input" defaultValue={this.state.text} placeholder='Join the conversation' />
                                        </div>
                                        <div className='button-div'>
                                            <button className='button add-comment-submit' type='submit'>Submit</button>
                                        </div>
                                    </form>
                                </div>
                            </div>
                        </article>
                    </div>
                    : <div className='box box-style box-style-comment box-style-comment-small'>
                        <article className='media'>
                            <div className='media-left'>
                                <img className='comment-form-image' src={this.props.user.avatar_url} />
                            </div>
                            <div className='media-content'>
                                <div className='content'>
                                    <div onClick={this.props.toggleForm.bind(this, null)} className='comment-form-small'>
                                        <p>Join the conversation</p>
                                    </div>
                                </div>
                            </div>
                        </article>
                    </div>
                }
            </div>
        );
    }
    handleChange(e) {
        this.setState({
            text: e.target.value
        });
    }
}

AddCommentForm.propTypes = {
    id: PropTypes.string.isRequired,
    addComment: PropTypes.func.isRequired,
    toggleForm: PropTypes.func.isRequired,
    user: PropTypes.object.isRequired,
    showForm: PropTypes.bool.isRequired,
    handleSubmit: PropTypes.func.isRequired
};

export default AddCommentForm;