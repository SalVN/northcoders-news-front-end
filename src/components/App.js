import React, {Component} from 'react';
import PropTypes from 'prop-types';

import TopicNavBar from './TopicNavBar';
import MainBanner from './MainBanner';
import UserCard from './UserCard';
import UserListCard from './UserListCard';

import './css/App.css';

export class App extends Component {
  render() {
    let pageStyling = window.innerWidth < 800 ? 'main-page-mobile' : 'main-page';
    return (
      <div className='page'>
        <TopicNavBar />
        <MainBanner />
        <div className={pageStyling}>
          <div className={'columns is-gapless'}>
            <div className='column is-three-quarters'>
              {this.props.children}
            </div>
            <div className='column is-one-quarter'>
              <div className='is-hidden-mobile'>
                <UserCard />
              </div>
              <UserListCard />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.element.isRequired
};

export default App;
