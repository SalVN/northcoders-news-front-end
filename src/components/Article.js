import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/Article.css';

const Article = function (props) {
    return (
        <div className='box'>
            <div className='columns'>
                <div className='column is-1'>
                    <button className='button is-black is-inverted is-small'><i className="fa fa-arrow-up" aria-hidden="true"></i></button>
                    <p className='comment-card-votes'><strong>{props.article.votes}</strong></p>
                    <button className='button is-black is-inverted is-small'><i className="fa fa-arrow-down" aria-hidden="true"></i></button>
                </div>
                <div className='column'>
                    <div className='title is-3 article-title'>{props.article.title}</div>
                    <p>
                        <span>
                            <span className='article-subheading-key'>added by:</span>
                            <Link to='#'>
                                <span>
                                    {props.user &&
                                        <img className='article-card-avatar' src={props.user.avatar_url} alt='user avatar' />
                                    }
                                </span>
                                <span className='article-subheading-value'>
                                    <strong>{props.article.created_by}</strong>
                                </span>
                            </Link>
                        </span>
                        <span>
                            <span className='article-card-subheading-key'>tags:</span>
                            <Link to={`/topics/${props.article.belongs_to}/articles`}>
                                <span className='article-card-subheading-value'>
                                    <strong>{props.article.belongs_to}</strong>
                                </span>
                            </Link>
                        </span>
                        <span>
                            <span className='article-card-subheading-key'>comments:</span>
                            {/*<Link to='#'>*/}
                            <span className='article-card-subheading-value-notlink'>
                                <strong>{props.article.commentCount}</strong>
                            </span>
                            {/*</Link>*/}
                        </span>
                    </p>
                    <div>{props.article.body}</div>
                </div>
            </div>
        </div>
    );
};

Article.propTypes = {
    article: PropTypes.object.isRequired,
    user: PropTypes.object.isRequired
};

export default Article;