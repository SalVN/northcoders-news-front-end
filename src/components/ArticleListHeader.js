import React from 'react';
import PropTypes from 'prop-types';

import './css/MainArticleList.css';

const ArticleListHeader = function (props) {
    return (
        <div className='content'>
            <div className='columns'>
                <div className='column is-3'>
                    <h2 className='page-title'><strong>{props.title}</strong></h2>
                </div>
                <div className='column is-1 is-offset-5 label'>
                    <label htmlFor="Sort by">Sort by:</label>
                </div>
                <div className='column is-3'>
                    <div className='select is-primary select-menu'>
                        <select name="Sort by" onChange={props.handleClickSelect}>
                            <option value="votes">Popularity</option>
                            <option value="comments">Number of Comments</option>
                            <option value="random">Random</option>
                        </select>
                    </div>
                </div>
            </div>
        </div>
    );
};

ArticleListHeader.propTypes = {
    title: PropTypes.string.isRequired,
    handleClickSelect: PropTypes.func.isRequired
};

export default ArticleListHeader;