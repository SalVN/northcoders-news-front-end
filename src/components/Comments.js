import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as actions from '../actions/actions';

import AddCommentForm from './AddCommentForm';
import CommentsList from './CommentsList';
import './css/Comments.css';

import { USERNAME } from '../../config';

export class Comments extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showForm: false,
            added: false,
            maximum: 8,
            voted: false
        };
        this.toggleForm = this.toggleForm.bind(this);
        this.deleteHandler = this.deleteHandler.bind(this);
        this.voteHandler = this.voteHandler.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.viewMoreComments = this.viewMoreComments.bind(this);
    }
    componentDidMount() {
        this.props.fetchComments(this.props.id);
        if (!this.props.user.username) {
            this.props.fetchUser(USERNAME);
        }
    }
    componentWillReceiveProps(newProps) {
        if ((newProps.comments && newProps.comments.length !== this.props.comments.length) || this.state.voted) {
            this.props.fetchUsers();
            this.props.fetchUser(USERNAME);
            this.setState({ voted: false });
        }
    }
    render() {
        return (
            <div id='comments' className='columns'>
                <div className='column is-one-quarter'>
                    <p className='comments-title'>
                        <span className='comments-title is-size-2-desktop is-size-4-tablet comments-title-main'>Comments</span>
                        {this.props.comments
                            ? <span className='comments-title-sub is-hidden-tablet-only comments-title-sub'>{`(${this.props.comments.length})`}</span>
                            : (<span className='comments-title-sub is-hidden-tablet-only'>{'(0)'}</span>)
                        }
                    </p>
                    <p className='is-hidden-mobile is-hidden-desktop'>
                        {this.props.comments
                            ? <span className='comments-title comments-title-sub tablet'>{`(${this.props.comments.length})`}</span>
                            : <span className='comments-title comments-title-sub tablet'>{'(0)'}</span>
                        }
                    </p>
                </div>
                <div className='column'>
                    {this.state.added &&
                        <div>
                            <h3>Your comment has been added</h3>
                        </div>
                    }
                    {this.props.commentsLoading
                        ? <div className='article-list-loading-icon'>
                            <span>
                                <i className='fa fa-refresh fa-spin' />
                            </span>
                        </div>
                        :
                        <div>
                            <AddCommentForm
                                user={this.props.user}
                                showForm={this.state.showForm}
                                toggleForm={this.toggleForm}
                                handleSubmit={this.handleSubmit}
                                id={this.props.id} />
                            <CommentsList
                                users={this.props.users}
                                comments={this.props.comments}
                                deleteHandler={this.deleteHandler}
                                voteHandler={this.voteHandler}
                                maximum={this.state.maximum}
                                viewMoreComments={this.viewMoreComments}
                            />
                        </div>}
                </div>
            </div>
        );
    }
    toggleForm(text) {
        if (!text || (text && text.length === 0)) {
            this.setState({
                showForm: !this.state.showForm,
                added: false
            });
        }
    }
    deleteHandler(id) {
        this.props.deleteComment(id);
    }
    voteHandler(vote, id) {
        this.props.voteComment(vote, id);
        this.setState({
            voted: true
        });
    }
    handleSubmit(e) {
        e.preventDefault();
        const obj = {
            body: e.target[0].value,
            created_by: USERNAME
        };
        this.props.addComment(obj, this.props.id);
        this.setState({
            added: true,
            showForm: false
        });
    }
    viewMoreComments() {
        const newMax = this.state.maximum + 10;
        this.setState({
            maximum: newMax
        });
    }
}

function mapDispatchToProps(dispatch) {
    return {
        fetchComments: (id) => {
            dispatch(actions.fetchComments(id));
        },
        deleteComment: (id) => {
            dispatch(actions.deleteComment(id));
        },
        voteComment: (vote, id) => {
            dispatch(actions.voteComment(vote, id));
        },
        fetchUser: (username) => {
            dispatch(actions.fetchOneUser(username));
        },
        addComment: (comment, id) => {
            dispatch(actions.addComment(comment, id));
        },
        fetchUsers: () => {
            dispatch(actions.fetchUsers());
        }
    };
}

function mapStateToProps(state) {
    return {
        comments: state.comments.comments,
        commentsLoading: state.comments.loading,
        userLoading: state.oneUser.loading,
        user: state.oneUser.user
    };
}

Comments.propTypes = {
    id: PropTypes.string.isRequired,
    comments: PropTypes.array.isRequired,
    users: PropTypes.array.isRequired,
    fetchComments: PropTypes.func.isRequired,
    deleteComment: PropTypes.func.isRequired,
    voteComment: PropTypes.func.isRequired,
    fetchUser: PropTypes.func.isRequired,
    user: PropTypes.object.isRequired,
    addComment: PropTypes.func.isRequired,
    fetchUsers: PropTypes.func.isRequired,
    commentsLoading: PropTypes.bool.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(Comments);