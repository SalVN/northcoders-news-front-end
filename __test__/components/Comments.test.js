import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import sinon from 'sinon';

const mockStore = configureStore();
const initialState = {};

import { Comments } from '../../src/components/Comments';

describe('Comments', () => {
    const users = [{
        _id: '59b1b18b327cce1fb043bdb1',
        username: 'northcoder',
        name: 'Awesome Northcoder',
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    }, {
        _id: '59b1b18b327cce1fb043bdb2',
        username: 'javascripter',
        name: 'Awesome Javascripter',
        avatar_url: 'https://avatars3.githubusercontent.com/u/6791502?v=3&s=200',
        __v: 0
    }];
    const comments = [{
        _id: '59b11ae18807841d9bf13234',
        body: 'this is a comment',
        belongs_to: '59b11ae18807841d9bf13232',
        __v: 0,
        created_by: 'northcoder',
        votes: 0,
        created_at: 1504778965845
    },
    {
        _id: '59b11ae18807841d9bf13235',
        body: 'this is another comment',
        belongs_to: '59b11ae18807841d9bf13232',
        __v: 0,
        created_by: 'northcoder',
        votes: 0,
        created_at: 1504778965845
    },
    {
        _id: '59b11ae18807841d9bf13236',
        body: 'this is my comment',
        belongs_to: '59b11ae18807841d9bf13232',
        __v: 0,
        created_by: 'northcoder',
        votes: 0,
        created_at: 1504778977306
    }];

    it('is a function', () => {
        expect(typeof Comments).toEqual('function');
    });

    it('renders', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(<Comments
            store={store}
            id={comments[0]._id}
            comments={comments}
            users={users}
            fetchComments={x => x}
            deleteComment={x => x}
            voteComment={x => x}
            fetchUser={x => x}
            user={users[0]}
            addComment={x => x}
            fetchUsers={x => x}
            commentsLoading={false}
        />);
        expect(wrapper.children().length).toEqual(2);
    });

    it('renders correctly', () => {
        const store = mockStore(initialState);
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <Comments
                        id={comments[0]._id}
                        comments={comments}
                        users={users}
                        fetchComments={x => x}
                        deleteComment={x => x}
                        voteComment={x => x}
                        fetchUser={x => x}
                        user={users[0]}
                        addComment={x => x}
                        fetchUsers={x => x}
                        commentsLoading={false}
                    />
                </MemoryRouter>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('fetches comments when the component mounts', () => {
        const spy = sinon.stub();
        const store = mockStore(initialState);
        mount(
            <MemoryRouter>
                <Comments
                    store={store}
                    id={comments[0]._id}
                    comments={comments}
                    users={users}
                    fetchComments={spy}
                    deleteComment={x => x}
                    voteComment={x => x}
                    fetchUser={x => x}
                    user={users[0]}
                    addComment={x => x}
                    fetchUsers={x => x}
                    commentsLoading={false}
                />
            </MemoryRouter>);
        expect(spy.callCount).toBe(1);
    });

    it('fetches the user when the component mounts if the data is not currently held in state', () => {
        const spy = sinon.stub();
        const store = mockStore(initialState);
        mount(
            <MemoryRouter>
                <Comments
                    store={store}
                    id={comments[0]._id}
                    comments={comments}
                    users={users}
                    fetchComments={x => x}
                    deleteComment={x => x}
                    voteComment={x => x}
                    fetchUser={spy}
                    user={users[0]}
                    addComment={x => x}
                    fetchUsers={x => x}
                    commentsLoading={false}
                />
            </MemoryRouter>);
        expect(spy.callCount).toBe(0);
        mount(
            <MemoryRouter>
                <Comments
                    store={store}
                    id={comments[0]._id}
                    comments={comments}
                    users={users}
                    fetchComments={x => x}
                    deleteComment={x => x}
                    voteComment={x => x}
                    fetchUser={spy}
                    user={{}}
                    addComment={x => x}
                    fetchUsers={x => x}
                    commentsLoading={false}
                />
            </MemoryRouter>);
        expect(spy.callCount).toBe(1);
    });

    it('fetches users and user if a comment is added', () => {
        const store = mockStore(initialState);
        const spyUsers = sinon.stub();
        const spyUser = sinon.stub();
        const wrapper = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={[]}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={spyUser}
                user={users[0]}
                addComment={x => x}
                fetchUsers={spyUsers}
                commentsLoading={true}
            />);
        expect(spyUser.callCount).toBe(0);
        expect(spyUsers.callCount).toBe(0);
        wrapper.setProps({ comments: [comments] });
        expect(spyUser.callCount).toBe(1);
        expect(spyUsers.callCount).toBe(1);
    });

    it('fetches users and user if a vote is made', () => {
        const store = mockStore(initialState);
        const spyUsers = sinon.stub();
        const spyUser = sinon.stub();
        const wrapper = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={comments}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={spyUser}
                user={users[0]}
                addComment={x => x}
                fetchUsers={spyUsers}
                commentsLoading={true}
            />);
        expect(spyUser.callCount).toBe(0);
        expect(spyUsers.callCount).toBe(0);
        wrapper.setState({ voted: true });
        wrapper.setProps({ comments: [comments] });
        expect(spyUser.callCount).toBe(1);
        expect(spyUsers.callCount).toBe(1);
    });

    it('has an initial state', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(<Comments
            store={store}
            id={comments[0]._id}
            comments={comments}
            users={users}
            fetchComments={x => x}
            deleteComment={x => x}
            voteComment={x => x}
            fetchUser={x => x}
            user={users[0]}
            addComment={x => x}
            fetchUsers={x => x}
            commentsLoading={false}
        />);
        expect(wrapper.state('showForm')).toBe(false);
        expect(wrapper.state('added')).toBe(false);
        expect(wrapper.state('maximum')).toBe(8);
        expect(wrapper.state('voted')).toBe(false);
    });

    it('renders the number of comments if a comments array is passed in props, or 0 if not', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(<Comments
            store={store}
            id={comments[0]._id}
            comments={comments}
            users={users}
            fetchComments={x => x}
            deleteComment={x => x}
            voteComment={x => x}
            fetchUser={x => x}
            user={users[0]}
            addComment={x => x}
            fetchUsers={x => x}
            commentsLoading={false}
        />);
        expect(wrapper.find('.comments-length').node.props.children).toBe('(3)');

        const wrapperNone = shallow(<Comments
            store={store}
            id={comments[0]._id}
            comments={null}
            users={users}
            fetchComments={x => x}
            deleteComment={x => x}
            voteComment={x => x}
            fetchUser={x => x}
            user={users[0]}
            addComment={x => x}
            fetchUsers={x => x}
            commentsLoading={false}
        />);
        expect(wrapperNone.find('.comments-length').node.props.children).toBe('(0)');
    });

    it('shows an added message if added is true in the state', () => {
        const store = mockStore(initialState);
        const wrapper = shallow(<Comments
            store={store}
            id={comments[0]._id}
            comments={comments}
            users={users}
            fetchComments={x => x}
            deleteComment={x => x}
            voteComment={x => x}
            fetchUser={x => x}
            user={users[0]}
            addComment={x => x}
            fetchUsers={x => x}
            commentsLoading={false}
        />);
        expect(wrapper.find('h3').length).toBe(0);
        wrapper.setState({ added: true });
        expect(wrapper.find('h3').node.props.children).toBe('Your comment has been added');
    });

    it('renders a loading spinner if usersLoading is given', () => {
        const store = mockStore(initialState);
        const wrapperA = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={comments}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={x => x}
                user={users[0]}
                addComment={x => x}
                fetchUsers={x => x}
                commentsLoading={true}
            />);
        expect(wrapperA.find('.fa-spin').length).toBe(1);

        const wrapperB = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={comments}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={x => x}
                user={users[0]}
                addComment={x => x}
                fetchUsers={x => x}
                commentsLoading={false}
            />);
        expect(wrapperB.find('.fa-spin').length).toBe(0);
    });

    it('renders the AddCommentForm and CommentsList if commentsLoading === false', () => {
        const store = mockStore(initialState);
        const wrapperA = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={[comments]}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={x => x}
                user={users[0]}
                addComment={x => x}
                fetchUsers={x => x}
                commentsLoading={true}
            />);
        expect(wrapperA.find('AddCommentForm').length).toBe(0);
        expect(wrapperA.find('CommentsList').length).toBe(0);

        const wrapperB = shallow(
            <Comments
                store={store}
                id={comments[0]._id}
                comments={comments}
                users={users}
                fetchComments={x => x}
                deleteComment={x => x}
                voteComment={x => x}
                fetchUser={x => x}
                user={users[0]}
                addComment={x => x}
                fetchUsers={x => x}
                commentsLoading={false}
            />);
        expect(wrapperB.find('AddCommentForm').length).toBe(1);
        expect(wrapperB.find('CommentsList').length).toBe(1);
    });
});