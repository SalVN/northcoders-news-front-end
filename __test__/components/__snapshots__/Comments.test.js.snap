// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders correctly 1`] = `
<div
  className="columns"
  id="comments"
>
  <div
    className="column is-one-quarter"
  >
    <p
      className="comments-title"
    >
      <span
        className="comments-title is-size-2-desktop is-size-4-tablet comments-title-main"
      >
        Comments
      </span>
      <span
        className="comments-title-sub is-hidden-tablet-only comments-title-sub"
      >
        (3)
      </span>
    </p>
    <p
      className="is-hidden-mobile is-hidden-desktop"
    >
      <span
        className="comments-title comments-title-sub tablet comments-length"
      >
        (3)
      </span>
    </p>
  </div>
  <div
    className="column"
  >
    <div>
      <div>
        <div
          className="box box-style box-style-comment box-style-comment-small"
        >
          <article
            className="media"
          >
            <div
              className="media-left"
            >
              <img
                className="comment-form-image"
                src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
              />
            </div>
            <div
              className="media-content"
            >
              <div
                className="content"
              >
                <div
                  className="comment-form-small"
                  onClick={[Function]}
                >
                  <p>
                    Join the conversation
                  </p>
                </div>
              </div>
            </div>
          </article>
        </div>
      </div>
      <div>
        <div
          className="comment-sort-links"
        >
          <a
            className="comment-sort-active"
            onClick={[Function]}
          >
            <span
              className="comment-sort-left"
            >
              sort by newest
            </span>
          </a>
          <a
            className="comment-sort-links"
            onClick={[Function]}
          >
            <span
              className="comment-sort-left"
            >
              sort by oldest
            </span>
          </a>
          <a
            className="comment-sort-links"
            onClick={[Function]}
          >
            <span>
              sort by votes
            </span>
          </a>
        </div>
        <div>
          <div
            className="card box-style"
          >
            <article
              className="media"
            >
              <div
                className="media-left"
              >
                <button
                  className="button is-black is-inverted is-small up"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-up"
                  />
                </button>
                <p
                  className="comment-card-votes"
                >
                  <strong>
                    0
                  </strong>
                </p>
                <button
                  className="button is-black is-inverted is-small down"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-down"
                  />
                </button>
              </div>
              <div
                className="media-content"
              >
                <div
                  className="content"
                >
                  <div>
                    <img
                      alt="user avatar"
                      className="comment-card-avatar"
                      src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
                    />
                    <a
                      href="/users/northcoder"
                      onClick={[Function]}
                    >
                      <span
                        className="comment-card-name"
                      >
                        <strong>
                          northcoder
                        </strong>
                      </span>
                    </a>
                    <p
                      className="comment-card-date"
                    >
                      7th September 2017, 10:09
                    </p>
                  </div>
                  <div
                    className="comment-card-body"
                  >
                    this is my comment
                  </div>
                  <button
                    className="button is-small delete-button"
                    onClick={[Function]}
                  >
                    Delete Comment
                  </button>
                </div>
              </div>
            </article>
          </div>
        </div>
        <div>
          <div
            className="card box-style"
          >
            <article
              className="media"
            >
              <div
                className="media-left"
              >
                <button
                  className="button is-black is-inverted is-small up"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-up"
                  />
                </button>
                <p
                  className="comment-card-votes"
                >
                  <strong>
                    0
                  </strong>
                </p>
                <button
                  className="button is-black is-inverted is-small down"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-down"
                  />
                </button>
              </div>
              <div
                className="media-content"
              >
                <div
                  className="content"
                >
                  <div>
                    <img
                      alt="user avatar"
                      className="comment-card-avatar"
                      src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
                    />
                    <a
                      href="/users/northcoder"
                      onClick={[Function]}
                    >
                      <span
                        className="comment-card-name"
                      >
                        <strong>
                          northcoder
                        </strong>
                      </span>
                    </a>
                    <p
                      className="comment-card-date"
                    >
                      7th September 2017, 10:09
                    </p>
                  </div>
                  <div
                    className="comment-card-body"
                  >
                    this is a comment
                  </div>
                  <button
                    className="button is-small delete-button"
                    onClick={[Function]}
                  >
                    Delete Comment
                  </button>
                </div>
              </div>
            </article>
          </div>
        </div>
        <div>
          <div
            className="card box-style"
          >
            <article
              className="media"
            >
              <div
                className="media-left"
              >
                <button
                  className="button is-black is-inverted is-small up"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-up"
                  />
                </button>
                <p
                  className="comment-card-votes"
                >
                  <strong>
                    0
                  </strong>
                </p>
                <button
                  className="button is-black is-inverted is-small down"
                  onClick={[Function]}
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-arrow-down"
                  />
                </button>
              </div>
              <div
                className="media-content"
              >
                <div
                  className="content"
                >
                  <div>
                    <img
                      alt="user avatar"
                      className="comment-card-avatar"
                      src="https://avatars3.githubusercontent.com/u/6791502?v=3&s=200"
                    />
                    <a
                      href="/users/northcoder"
                      onClick={[Function]}
                    >
                      <span
                        className="comment-card-name"
                      >
                        <strong>
                          northcoder
                        </strong>
                      </span>
                    </a>
                    <p
                      className="comment-card-date"
                    >
                      7th September 2017, 10:09
                    </p>
                  </div>
                  <div
                    className="comment-card-body"
                  >
                    this is another comment
                  </div>
                  <button
                    className="button is-small delete-button"
                    onClick={[Function]}
                  >
                    Delete Comment
                  </button>
                </div>
              </div>
            </article>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
